### ISSUE TITLE
Diesel server status 500

### ISSUE THREAD

**Issue by 4ydan** [ElektraInitiative/PermaplanT#758]
When creating a map with a name that already exists the server responds with 500.
![image](https://github.com/ElektraInitiative/PermaplanT/assets/53004139/bf905a0e-d86e-4f69-8bd7-c14b882aa561)

**Comment by markus2330** [comment]
Thank you for reporting this problem!

**Comment by badnames** [comment]
Just FYI if you look at the server response it should say what exactly went wrong.
It might be useful to include this information into this bug report.

**Comment by 4ydan** [comment]
Yes thank you. The message is fine `duplicate key value violates unique constraint "map_unique_name"`, its just that the error code is not appropriate.

**Comment by markus2330** [comment]
The user-facing error message is also not ideal. There should be a different status code for duplicated maps, so that we can be sure. (And don't need to say "maybe" it is a duplicated map).

A better solution would be to directly validate the map name while typing, though.

**Comment by markus2330** [comment]
@tschawax do we need an extra endpoint to check for duplicate map names or can we have better usability by simply having a different return code for duplicate map names?

**Comment by markus2330** [comment]
Ok, `409 Conflict` sounds reasonable, I updated the top-post.

> error/info message

I would prefer if the user would not get a toast but see a message directly at the problematic field, ideally while typing.
